programs/shfmt.nix

```
@@ -7,8 +7,8 @@ in
     enable = lib.mkEnableOption "shfmt";
     package = lib.mkPackageOption pkgs "shfmt" { };
     indent_size = lib.mkOption {
-      type = lib.types.int;
-      default = 2;
+      type = lib.types.nullOr lib.types.int;
+      default = null;
```

Changing the default would make this a breaking change for the users.

```suggestion
      default = 2;
```

---

Let's do the non-breaking change as treefmt-nix has no release mechanism at the moment. Users expect the latest main to be relatively stable.

You can use the option description field to describe shfmt's behaviour.

To do the breaking change, I would first add a nix eval trace to warn existing users that the default is going to change, and tell then what to do. Then wait a couple of months and change the default then.